File: ecommerce/src/main/java/com/june/ecommerce/domain/seller/Seller.java
package com.june.ecommerce.domain.seller;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;

import java.time.LocalDateTime;

@Entity
@Getter
public class Seller {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;
    private String description;
    private String logoUrl;
    private Double rating;
    private String contactEmail;
    private String contactPhone;

    private LocalDateTime createAt =LocalDateTime.now();
}

File: src/domain/entities/Seller.ts
import User from "./User";

export default class Seller extends User {
  constructor(
    public name: string,
    public description: string,
    public logo_url: string,
    public rating: number,
    public contact_email: string,
    public contact_phone: string,
    public id?: number,
  ) {
    super(id);
  }
}
